bpfctl/src/main.rs

```diff
@@ -874,23 +888,70 @@ async fn execute_request_tcp(
 async fn execute_request(command: &Commands, channel: Channel) -> anyhow::Result<()> {
     let mut client = BpfdClient::new(channel);
     match command {
-        Commands::LoadFromFile(_) | Commands::LoadFromImage(_) => {
-            let attach_info = match command.get_attach_info() {
+        Commands::LoadFromFile(l) => {
+            let bytecode = match command.get_bytecode_location() {
+                Ok(t) => t,
+                Err(e) => bail!(e),
+            };
+
+            let attach = match command.get_attach_info() {
                 Ok(t) => t,
                 Err(e) => bail!(e),
             };
 
-            let common = match command.get_request_common() {
+            let request = tonic::Request::new(LoadRequest {
+                bytecode,
+                name: l.name.to_string(),
+                program_type: l.command.get_prog_type() as u32,
+                attach,
+                metadata: l
+                    .metadata
+                    .clone()
+                    .unwrap_or(vec![])
+                    .iter()
+                    .map(|(k, v)| (k.to_owned(), v.to_owned()))
+                    .collect(),
+                global_data: parse_global(&l.global),
+                uuid: None,
+                map_owner_id: l.map_owner_id,
+            });
+            let response = client.load(request).await?.into_inner();
+
+            ProgTable::new_get_bpfd(&response.info)?.print();
+            ProgTable::new_get_unsupported(&response.kernel_info)?.print();
+        }
+
+        Commands::LoadFromImage(l) => {
+            let bytecode = match command.get_bytecode_location() {
+                Ok(t) => t,
+                Err(e) => bail!(e),
+            };
+
+            let attach = match command.get_attach_info() {
                 Ok(t) => t,
                 Err(e) => bail!(e),
             };
 
             let request = tonic::Request::new(LoadRequest {
-                common,
-                attach_info,
+                bytecode,
+                name: l.name.to_string(),
+                program_type: l.command.get_prog_type() as u32,
+                attach,
+                metadata: l
+                    .metadata
+                    .clone()
+                    .unwrap_or(vec![])
+                    .iter()
+                    .map(|(k, v)| (k.to_owned(), v.to_owned()))
+                    .collect(),
+                global_data: parse_global(&l.global),
+                uuid: None,
+                map_owner_id: l.map_owner_id,
             });
             let response = client.load(request).await?.into_inner();
-            println!("{}", response.id);
+
+            ProgTable::new_get_bpfd(&response.info)?.print();
+            ProgTable::new_get_unsupported(&response.kernel_info)?.print();
         }
```

Comment:
Both patterns here are doing the exact same thing can you just do something like this?

```suggestion
        _(l) => {
            let bytecode = match command.get_bytecode_location() {
                Ok(t) => t,
                Err(e) => bail!(e),
            };

            let attach = match command.get_attach_info() {
                Ok(t) => t,
                Err(e) => bail!(e),
            };

            let request = tonic::Request::new(LoadRequest {
                bytecode,
                name: l.name.to_string(),
                program_type: l.command.get_prog_type() as u32,
                attach,
                metadata: l
                    .metadata
                    .clone()
                    .unwrap_or(vec![])
                    .iter()
                    .map(|(k, v)| (k.to_owned(), v.to_owned()))
                    .collect(),
                global_data: parse_global(&l.global),
                uuid: None,
                map_owner_id: l.map_owner_id,
            });
            let response = client.load(request).await?.into_inner();

            ProgTable::new_get_bpfd(&response.info)?.print();
            ProgTable::new_get_unsupported(&response.kernel_info)?.print();
        }
```

Reply:
I agree on the duplicate code, but stuck on how to do it in rust.